services/brig/src/Brig/API.hs

```diff
@@ -1200,42 +1200,63 @@ createUserNoVerify uData = do
 
 deleteUserNoVerifyH :: UserId -> Handler Response
 deleteUserNoVerifyH uid = do
+    setStatus status202 empty <$ deleteUserNoVerify uid
+
+deleteUserNoVerify :: UserId -> Handler ()
+deleteUserNoVerify uid = do
     void $ lift (API.lookupAccount uid) >>= ifNothing userNotFound
     lift $ API.deleteUserNoVerify uid
-    return $ setStatus status202 empty
 
 changeSelfEmailNoSendH :: UserId ::: JsonRequest EmailUpdate -> Handler Response
 changeSelfEmailNoSendH (u ::: req) = changeEmail u req False
 
 checkUserExistsH :: UserId ::: UserId -> Handler Response
 checkUserExistsH (self ::: uid) = do
-    exists <- lift $ isJust <$> API.lookupProfile self uid
+    exists <- lift $ checkUserExists self uid
     if exists then return empty else throwStd userNotFound
 
+checkUserExists :: UserId -> UserId -> AppIO Bool
+checkUserExists self uid = do
+    isJust <$> API.lookupProfile self uid
+
 getSelfH :: JSON ::: UserId -> Handler Response
 getSelfH (_ ::: self) = do
-    p <- (lift $ API.lookupSelfProfile self) >>= ifNothing userNotFound
-    return $ json p
+    json <$> getSelf self
+
+getSelf :: UserId -> Handler SelfProfile
+getSelf self = do
+    lift (API.lookupSelfProfile self) >>= ifNothing userNotFound
 
 getUserH :: JSON ::: UserId ::: UserId -> Handler Response
 getUserH (_ ::: self ::: uid) = do
-    p <- (lift $ API.lookupProfile self uid) >>= ifNothing userNotFound
-    return $ json p
+    json <$> getUser self uid
+
+getUser :: UserId -> UserId -> Handler UserProfile
+getUser self uid = do
+    lift (API.lookupProfile self uid) >>= ifNothing userNotFound
 
 getUserNameH :: JSON ::: UserId -> Handler Response
 getUserNameH (_ ::: self) = do
-    name <- lift $ API.lookupName self
+    name :: Maybe Name <- lift $ API.lookupName self
     return $ case name of
         Just n  -> json $ object ["name" .= n]
         Nothing -> setStatus status404 empty
 
 listUsersH :: JSON ::: UserId ::: Either (List UserId) (List Handle) -> Handler Response
-listUsersH (_ ::: self ::: qry) = case qry of
-    Left  us -> toResponse =<< byIds (fromList us)
+listUsersH (_ ::: self ::: qry) =
+    toResponse <$> listUsers self qry
+  where
+    toResponse = \case
```

Comment:
Not necessary to change, just wanted to note that this could maybe be done more nicely by using non-empty lists, e.g. `maybe (setStatus status404 empty) json . nonEmpty <$> listUsers self qry`.

Reply:
still, good idea.  *arguably* nicer, yes, but then we need to worry about the serialization of non-empty lists being compatible with the one for empty list.